### PR TITLE
Add Microsoft Teams notification

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
## What 
I added `microsoftteams` field for [Microsoft Teams](https://spinnaker.io/setup/features/notifications/#microsoft-teams) notification configuration which was added in the Spinnaker `1.23.0` release. 

The fields are same as the `halconfig` described  [here](https://spinnaker.io/setup/features/notifications/#microsoft-teams). 
Thanks in advance.

## Why
It was missing. 
Described in [Microsoft Teams vs Slack: Which Collaboration App Is Better?](https://kinsta.com/blog/microsoft-teams-vs-slack/), there are many Microsoft Teams adoption in the world. We should add them soon.

## Ref

* Document
    * [Spinnaker Adds Microsoft Teams Notification Support, Armory](https://www.armory.io/blog/spinnaker-adds-microsoft-teams-notification-support/)
    * [Version 1.23, Spinnaker](https://spinnaker.io/community/releases/versions/1-23-1-changelog)
* Related PR
    * [feat(notification): Add Microsoft Teams notification support, spinnaker/deck](https://github.com/spinnaker/deck/pull/8503)

### CODE REVIEW

File: docs/docs.md [hunk 1]
```diff
@@ -4932,6 +4935,37 @@ Configuration for Google Chat notifications.
...
+ 
+
+ 
+
+ 
+
+ 
+
+
+
```
Comment (KeisukeYamashita) [inline docs/docs.md:4968]: [note]
Theses are auto generated by the `kleat-protobuilder`. I ran `make proto` to generate these fields.

**Review by ezimanyi — APPROVED** [review]
This looks great, thanks for adding this!

**Comment by maggieneterval** [comment]
@KeisukeYamashita I think we need to also update the [echo](https://github.com/spinnaker/kleat/blob/master/api/proto/config/echo.proto#L17) message and [translate](https://github.com/spinnaker/kleat/blob/master/internal/convert/echo.go#L27
) the `microsoftteams` block from the halconfig  to the echo config, right?

**Comment by KeisukeYamashita** [comment]
@maggieneterval Thanks for your quick review🙇‍♂️ Yeah, I totally missed that point, I will raise another PR🙇‍♂️

**Pull request merged**